src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -846,11 +896,35 @@ private void CheckIncrementDecrementSignature(BindingDiagnosticBag diagnostics)
             diagnostics.Add(this.GetFirstLocation(), useSiteInfo);
         }
 
-        private bool MatchesContainingType(TypeSymbol type)
+        private bool MatchesContainingType(TypeSymbol type, bool checkStrippedType)
         {
+            if (ContainingType is { IsExtension: true, ExtensionParameter.Type: var extendedType })
+            {
+                if (extendedType is null)
+                {
+                    return true; // An error scenario
+                }
+
+                if (checkStrippedType && type.IsNullableType())
```

Comment:
We should keep track of this and include in a spec update. Also applies to the proposed validation rules (marked PROTOTYPE in other file)

Reply:
> We should keep track of this and include in a spec update

We certainly can re-confirm the behavior, but it follows the current speclet, which doesn't prevent declaration of operators for ```Nullable<T>```. The other place  (marked PROTOTYPE in other file) is adding restrictions that are not mentioned in the speclet